src/GHC.hs

```diff
@@ -204,7 +102,7 @@ stage2Packages = return [haddock]
 -- 'Library', the function simply returns its name.
 programName :: Context -> String
 programName Context {..}
-    | package == ghc      = "ghc-stage" ++ show (fromEnum stage + 1)
+    | package == ghc      = "ghc"
```

Comment:
After this simplification we can probably just grab the name of the executable from the Cabal file.